src/com/carolinarollergirls/scoreboard/defaults/DefaultRulesetsModel.java

```diff
@@ -0,0 +1,339 @@
+            new IntegerRule(Penalty.SETTING_FO_LIMIT, "After how many penalties a skater has fouled out of the game. Note that the software currently does not support more than 9 penalties per skater.", 7),
+        };
+        Map<String, String> rootSettings = new HashMap<String, String>();
+        for (Rule r : knownRules) {
+            rules.put(r.getFullName(), r);
+            rootSettings.put(r.getFullName(), r.getDefaultValue());
+        }
+        // The root ruleset is always created from the above list,
+        // so that as rules are added over time there will always
+        // be a value for them.
+        RulesetModel root = addRuleset("WFTDA Sanctioned", "", rootId);
+        root.setAll(rootSettings);
+    }
+
+    public void reset() {
+        synchronized (coreLock) {
+            setCurrentRuleset(rootId);
+        }
+    }
+
+    public String getId() {
+        synchronized (coreLock) {
+            return id;
+        }
+    }
+    public void setId(String i) {
+        synchronized (coreLock) {
+            id = i;
+            scoreBoardChange(new ScoreBoardEvent(this, EVENT_CURRENT_RULESET, "", ""));
+        }
+    }
+    public String getName() {
+        synchronized (coreLock) {
+            return name;
+        }
+    }
+    public void setName(String n) {
+        synchronized (coreLock) {
+            name = n;
+            scoreBoardChange(new ScoreBoardEvent(this, EVENT_CURRENT_RULESET, "", ""));
+        }
+    }
+    public void setCurrentRuleset(String id) {
+        synchronized (coreLock) {
+            current.clear();
+            setCurrentRulesetRecurse(id);
+            scoreBoardChange(new ScoreBoardEvent(this, EVENT_CURRENT_RULESET, "", ""));
+        }
+    }
+
+    private void setCurrentRulesetRecurse(String i) {
+        Ruleset rs = getRuleset(i);
+        if (!rs.getId().equals(rootId)) {
+            setCurrentRulesetRecurse(rs.getParentRulesetId());
+        }
+        current.putAll(rs.getAll());
+        name = rs.getName();
+        id = rs.getId();
+    }
+
+    public Map<String, String> getAll() {
+        synchronized (coreLock) {
+            return Collections.unmodifiableMap(current);
+        }
+    }
+    public String get(String k) {
+        synchronized (coreLock) {
+            return current.get(k);
+        }
+    }
+    public boolean getBoolean(String k) {
+        synchronized (coreLock) {
+            return Boolean.parseBoolean(get(k));
+        }
+    }
+    public int getInt(String k) {
+        synchronized (coreLock) {
+            return Integer.parseInt(get(k));
+        }
+    }
+    public long getLong(String k) {
+        synchronized (coreLock) {
+            return Long.parseLong(get(k));
+        }
+    }
+    public void set(String k, String v) {
+        synchronized (coreLock) {
+            Rule r = rules.get(k);
+            if (r == null || !r.isValueValid(v)) {
+                return;
+            }
+            current.put(k, v);
```

Comment:
When doing this there should be some kind of indication in `name` and possibly `id` that the ruleset has been modified. (Maybe add a pre- or suffix "(modified)")

Reply:
At the very least, there should be a flag that indicates that the ruleset has been changed. Even if the frontend doesn't use the functionality yet, it's never a good idea to have such unwritten requirements for future changes.

Reply:
There is no functionality for the frontend to use, this code is only called when loading the save.